dynamic_island_watermark/lib/dynamic_island_watermark.dart

```diff
@@ -132,6 +139,22 @@ class _IslandWatermarkState extends State<_IslandWatermark> {
       _ => null,
     };
   }
+
+  bool _isPortrait() {
+    final mediaQuery = MediaQuery.maybeOf(context);
+    if (mediaQuery != null) {
+      return mediaQuery.orientation == Orientation.portrait;
+    }
+
+    final view = View.maybeOf(context);
+    if (view != null) {
+      final size = view.physicalSize;
+      return size.height >= size.width;
+    }
+
+    // Default to portrait if we cannot determine orientation safely.
+    return true;
```

Comment:
Shouldn't we default to not show it, since we might show it in a user visible place?